ReactApplication/src/Components/ColorfulGrid.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState } from "react";
+import { SketchPicker } from "react-color";
+import "./style.css";
+import { getRandomColor } from "../Helpers/Helpers";
+
+const INITIAL_COLORS = getRandomColor(5);
+const ColorfulGrid = () => {
+  const [color, setColor] = useState("#ffffff");
+  const [colors, setColors] = useState(INITIAL_COLORS);
+  const [rows, setRows] = useState(3);
+  const [columns, setColumns] = useState(3);
+
+  // handlers
+  function onDelete(index) {
+    setColors((prev) => {
+      let remainColors = [];
+      remainColors = prev.filter((c, i) => i !== index);
+      return remainColors;
+    });
+  }
+
+  function colorChanhgeHandler(index, newColor) {

```

Comment:
```suggestion
  function colorChangeHandler(index, newColor) {

```
do make sure to avoid typos in the code. you can add extention in VS code for the same.